hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -832,7 +832,7 @@ private void finishActiveMasterInitialization(MonitoredTask status) throws IOExc
             HBaseFsck.createLockRetryCounterFactory(this.conf).create());
       } finally {
         if (result != null) {
-          IOUtils.closeQuietly(result.getSecond());
+          Closeables.close(result.getSecond(), true);
```

Comment:
What is the difference about Closeables.close and IOUtils.close?

Reply:
Closeables.close(c, true) will swallow the IOException thrown while IOUtils.close will not.

Reply:
Seems there are three methods you used: Closeables.closeQuietly, Closeables.close, IOUtils.closeQuietly. Do we have a developer guide doc about what we recommend to use?

Reply:
Closeables.close will still throw IOExceptions on the method signature, so usually it is only suitable to be used in UT, where we are free to change the method signature.
IOUtils.closeQuietly will not throw any exceptions but you need to pass in a consumer to process the IOException(usually just a logging).
Closeables.closeQuietly is for closing Reader and InputStream, where we will not throw any exceptions.

Reply:
IOUtils.closeQuietly is also deprecated with a long time -- https://github.com/apache/commons-io/commit/ce2b3e87bfd15ba14b9e0c162e2b9c9549e1f3a8

Reply:
As is Closeables#closeQuietly... https://github.com/google/guava/issues/1118

Reply:
For IOUtils.closeQuietly, now it provides a version with a consumer parameter to process the IOException which is not deprecated.
For Closeables.closeQuirtly, the version which takes a Closeable as parameter has already been removed for a long time, but the version which takes a InputStream or a Reader is still available and is not deprecated.